projects/stargazer/plugins/other/radius/server.cpp

```diff
@@ -55,36 +55,35 @@ RadProto::Packet Server::makeResponse(const RadProto::Packet& request)
 
 void Server::handleSend(const error_code& ec)
 {
-    if (!m_token.stop_requested())
-        startReceive();
+    if (m_token.stop_requested())
+        return;
 
     if (ec)
     {
         m_logger("Error asyncSend: %s", ec.message());
         printfd(__FILE__, "Error asyncSend: '%s'\n", ec.message());
     }
+    startReceive();
 }
 
 void Server::handleReceive(const error_code& error, const std::optional<RadProto::Packet>& packet, const boost::asio::ip::udp::endpoint& source)
 {
-    if (!m_token.stop_requested())
+    if (m_token.stop_requested())
+        return;
+
+    if (error)
     {
-        if (error)
-        {
-            m_logger("Error asyncReceive: %s", error.message());
-            printfd(__FILE__, "Error asyncReceive: '%s'\n", error.message());
-            return;
-        }
+        m_logger("Error asyncReceive: %s", error.message());
+        printfd(__FILE__, "Error asyncReceive: '%s'\n", error.message());
+        return;
```

Comment:
It's interesting. In `handleSend` on error you log it and proceed. In `handleReceive` on error you log it and stop the server. Error handling should be uniform, you either proceed or stop. And since this is a server and it should be robust, it is better to proceed. Please make changes accordingly.